src/main/java/com/automation/school/ad_main/Collection/CompanyOrder.java

```diff
@@ -0,0 +1,87 @@
+package com.automation.school.ad_main.Collection;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompanyOrder implements Order{
+
+
+    private long id; //(generate automatically in constructor )
+    private LocalDateTime date; //(set as current day directly in constructor)
+    private User user;
+    public List<Company> companies;
+    private static int nextUniqueId = 0;
+    private int itemCount = 0;
+    Double value;
+
+    // public  User newUser;
+    //public  Item newItem;
+    //get the value of id
+    public CompanyOrder() {
+        this.date = LocalDateTime.now();
+        this.companies = new ArrayList<>(itemCount);
```

Comment:
why do you create a zero-length list?

Reply:
I understand that initializes the list but doesn't add anything to the list, so the list size is 0.
and i have the same contractions in the new code.  
 public List<Item> itemsCompany; - initializes .

 public CompanyOrder() {
        companyCurrentOrderId++;
        this.date = LocalDateTime.now();
        this.itemsCompany = new LinkedList<>();  // create empty list for current object 
    }
and next method
public void addItem(Item items) {
        itemsCompany = new LinkedList<>();
        itemsCompany.add(items);
    }

I think this approach resolve problem but it add only one last item in list. i will think how to resolve. 
public void addItem(Item items) {
        itemsCompany = new LinkedList<>();
        itemsCompany.add(items);
    }